lib/iris/analysis/stats.py

```diff
@@ -94,7 +95,29 @@ def _get_calc_view(cube_a, cube_b, corr_coords):
     reshaped_b = data_b.transpose(slice_ind+res_ind)\
                        .reshape(dim_i_len, dim_j_len)
 
-    return reshaped_a, reshaped_b, res_ind
+    # Remove data where one or both cubes are masked
+    # First deal with the case that either cube is unmasked
+    # Collapse masks to the dimension we are correlating over (0th)
+    if np.ma.is_masked(reshaped_a):
```

Comment:
Should be just `ma.is_masked(reshaped_a)` with an additional import at the top of the file `import numpy.ma as ma`. This will match the style used elsewhere.